test/distributed/test_c10d.py

```diff
@@ -4218,100 +4218,121 @@ def forward(self, inp):
+            model.zero_grad(set_to_none=False)
+            ddp_model.zero_grad(set_to_none=False)
+            self._train_model(model, input, target, loss, run_checkpoint=run_checkpoint)
+            self._train_model(ddp_model, ddp_input, ddp_target, loss, run_checkpoint=run_checkpoint)
+            for i, j in zip(model.parameters(), ddp_model.parameters()):
+                self.assertTrue(i.grad is not None)
+                self.assertTrue(j.grad is not None)
+                self.assertEqual(i.grad, j.grad)
 
     # DDP works as expect when layer is checkpointed only once
     @requires_nccl()
-    @skip_if_not_multigpu
+    @skip_if_lt_x_gpu(2)
     def test_ddp_checkpointing_once(self):
         store = c10d.FileStore(self.file_name, self.world_size)
         process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
-        for use_bucket_view in (True, False):
-            model = self._test_ddp_checkpointing(checkpoint_once=True,
-                                                 process_group=process_group,
-                                                 use_bucket_view=use_bucket_view)
-            norm = 0.0
-            for p in model.parameters():
-                self.assertTrue(p.grad is not None)
-                norm += p.grad.norm().item()
-            assert numpy.allclose(norm, 78053), norm
+        for use_bucket_view, static_graph in product((False, True), (False, True)):
+            self._test_ddp_checkpointing(self.CheckpointOnceModule(),
+                                         process_group=process_group,
+                                         use_bucket_view=use_bucket_view,
+                                         static_graph=static_graph)
 
     # DDP will fail when there are unused_parameters in the model
     @requires_nccl()
-    @skip_if_not_multigpu
+    @skip_if_lt_x_gpu(2)
     def test_ddp_checkpointing_unused_params(self):
         store = c10d.FileStore(self.file_name, self.world_size)
         process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
         for use_bucket_view in (True, False):
-            with self.assertRaisesRegex(
-                RuntimeError,
-                "Expected to mark a variable ready only once.",
-            ):
-                model = self._test_ddp_checkpointing(checkpoint_once=True,
-                                                     process_group=process_group,
-                                                     use_bucket_view=use_bucket_view,
-                                                     find_unused_parameters=True)
+            model = self._test_ddp_checkpointing(self.CheckpointOnceModule(),
+                                                 process_group=process_group,
+                                                 use_bucket_view=use_bucket_view,
+                                                 find_unused_parameters=True,
+                                                 static_graph=True)
```

Comment:
Should we keep the previous test that also tested with no static graph? 

And for the error, can we add string to recommend `use_static_graph` if it works for their use case to error message?

Reply:
sounds good!